pylint/checkers/base/basic_checker.py

```diff
@@ -657,13 +657,38 @@ def _check_misplaced_format_function(self, call_node: nodes.Call) -> None:
             ):
                 self.add_message("misplaced-format-function", node=call_node)
 
+    @staticmethod
+    def _is_terminating_func(node: nodes.Call) -> bool:
+        """Detect call to exit(), quit(), os._exit(), or sys.exit()."""
+        if (
+            not isinstance(node.func, nodes.Attribute)
+            and not (isinstance(node.func, nodes.Name))
+            or isinstance(node.parent, nodes.Lambda)
+        ):
+            return False
+
+        qnames = {"_sitebuiltins.Quitter", "sys.exit", "posix._exit", "nt._exit"}
+
+        try:
+            for inferred in node.func.infer():
+                if inferred.qname() in qnames:
```

Comment:
Not all types returned by `infer()` have a `qname()`. We should look for a specific type of `node` with `isinstance` and then use `qname`.